src/rgw/rgw_lc.cc

```diff
@@ -589,17 +596,59 @@ static int remove_expired_obj(const DoutPrefixProvider *dpp, lc_op_ctx& oc, bool
     return ret;
   }
 
-  obj = bucket->get_object(obj_key);
-  std::unique_ptr<rgw::sal::Object::DeleteOp> del_op = obj->get_delete_op(&oc.rctx);
+  // XXXX currently, rgw::sal::Bucket.owner is always null here
+  if (! bucket->get_owner()) {
+    auto& bucket_info = bucket->get_info();
+    std::unique_ptr<rgw::sal::User> user = store->get_user(bucket_info.owner);
+    // forgive me, lord
+    if (user) {
+      bucket->set_owner(user.get());
+    }
+  }
```

Comment:
Right here, the owner pointer is now junk, since `user` went out of scope.  At this point you can't use the owner pointer anymore.  It will be non-null, but point to freed memory.

Reply:
oh my

Reply:
fixed